src/kani-compiler/rustc_codegen_kani/src/codegen/intrinsic.rs

```diff
@@ -963,4 +964,30 @@ impl<'tcx> GotocCtx<'tcx> {
             .collect();
         self.codegen_expr_to_place(p, Expr::vector_expr(cbmc_ret_ty, elems))
     }
+
+    /// A volatile write of a memory location:
+    /// https://doc.rust-lang.org/std/ptr/fn.write_volatile.html
+    ///
+    /// Undefined behavior if any of these conditions are violated:
+    ///  * `dst` must be valid for writes (done by `--pointer-check`)
+    ///  * `dst` must be properly aligned (done by `align_check` below)
+    fn codegen_volatile_store(
+        &mut self,
+        instance: Instance<'tcx>,
+        intrinsic: &str,
+        mut fargs: Vec<Expr>,
+        loc: Location,
+    ) -> Stmt {
+        warn!(
+            "Kani does not support concurrency for now. {} treated as a sequential operation.",
+            intrinsic
```

Comment:
Can we dump a srcloc too, to help the user figure out where this happened?

Reply:
Yes, I will also wrap it as a util function because we are using this message in many places now.

Reply:
Done!